### PR TITLE
Grab multiple rows from a database

### LINKED ISSUE #3813: Grab multiple rows from a database
I'm trying to add some object with parameters which depends on some tables in a database. For example, let's call the object – 'car' and one of parameters call 'type'. The 'type' can't be managed from UI, it's already filled in the database. When I generate the 'car' I should take one of parameters from the database. Of course, I can look at the database manually and write my faker method to generate the 'type' but it's not very good solution because that data can be changed. In this case, I should take all available types from the database and select one of them.

Unfortunately, the Codeception don't have a method to get more than one row from a database. If I use `$I->grabFromDatabase('types', 'id');`, I'll always get the first row. I write my own method in new module for getting multiple rows, but I think it would be great to have this method in the Codeception 'from the box'.

If you need more abstract reason, I can tell the case by another words: when we should generate an object for test(s) but data of this object is depends from outside source (that can be changed) we must generate fake data using this source without duplicating this in the code.

### PR DESCRIPTION
**Pull request by RebOOter**
Closes #3813

### CODE REVIEW

**Comment by RebOOter** [comment]
Tests failed by this error:
`[GuzzleHttp\Exception\ConnectException] cURL error 7: Failed to connect to localhost port 8010: Connection refused (see http://curl.haxx.se/libcurl/c/libcurl-errors.html)`
I don't think that the reason is in commit (

**Comment by RebOOter** [comment]
@Naktibalda Please, could you give me a list of names or emails in the test database? I need to compare column in the test method

**Comment by Naktibalda** [comment]
https://github.com/Codeception/Codeception/blob/2.3/tests/data/dumps/sqlite.sql

**Comment by RebOOter** [comment]
@Naktibalda Sorry, but i don't understand why tests failed after last commit. In the previous commit (2b1fcfc) actual array contains another two emails: 
```
+    4 => 'user@example.org'
+    5 => 'user2@example.org'
```
I added this emails for assert but now in logs I see, that:
```
-    4 => 'user@example.org'
-    5 => 'user2@example.org'
``` 
Please, could you give me advice and tell how I can fix this?

**Comment by Naktibalda** [comment]
It happens because postgresql dump inserts 2 additional users: https://github.com/Codeception/Codeception/blob/2.3.3/tests/data/dumps/postgres.sql#L412-L433

**Comment by RebOOter** [comment]
@Naktibalda Yes, I know, this users (emails) in array for assert, I added them in this commit (2252d8c), but test doesn't work.
```
$this->assertEquals([
            'davert@mail.ua',
            'nick@mail.ua',
            'miles@davis.com',
            'charlie@parker.com',
            'user@example.org',
            'user2@example.org'
                                ],
            $emails);
```

**Comment by Naktibalda** [comment]
PostgresqlTest passed, but SqliteTest failed.

**Comment by RebOOter** [comment]
@Naktibalda Oh, you right, now I see, sorry. I think there are two solutions for make test:
1. Write switch/condition construction in TestsForDb.
2. Write different methods in MySqlDbTest, PostgreSqlDbTest and in SqliteDbTest.

Which solution do you prefer?

**Comment by Naktibalda** [comment]
`2.` I think that testing with one driver should be enough, there is nothing driver specific in your method.

**Comment by RebOOter** [comment]
@Naktibalda I think I'm done here)

File: tests/unit/Codeception/Module/Db/MySqlDbTest.php [hunk 1]
```diff
@@ -68,4 +68,16 @@ public function testConnectionIsResetOnEveryTestWhenReconnectIsTrue()
         $this->assertNotEquals($connection3, $connection2);
     }
 
+    public function testGrabColumnFromDatabase()
+    {
+        $emails = $this->module->grabColumnFromDatabase('users', 'email');
+        $this->assertEquals(
+            ['davert@mail.ua',
```
Comment (Naktibalda) [inline tests/unit/Codeception/Module/Db/MySqlDbTest.php]: Please move the first element of array to the next line.

File: src/Codeception/Module/Db.php [hunk 2]
```diff
@@ -517,6 +517,19 @@ protected function proceedSeeInDatabase($table, $column, $criteria)
         return $sth->fetchColumn();
     }
 
+    public function grabColumnFromDatabase($table, $column, array $criteria = [])
```
Comment (DavertMik) [inline src/Codeception/Module/Db.php]: Please add documentation in docblock for this method.
The similar method `grabFromDatabase` is documented in its interface https://github.com/Codeception/Codeception/blob/2.3/src/Codeception/Lib/Interfaces/Db.php
Use its documentation as an example and write some docs here. Usage examples with code would be nice.

File: src/Codeception/Module/Db.php [hunk 3]
```diff
@@ -517,6 +517,34 @@ protected function proceedSeeInDatabase($table, $column, $criteria)
         return $sth->fetchColumn();
     }
 
+    /**
+     * Fetches all values from the column in database.
+     * Provide table name, desired column and criteria.
+     *
+     * ``` php
+     * <?php
+     * $mails = $I->grabColumnFromDatabase('users', 'name', array('name' => 'RebOOter'));
```
Comment (DavertMik) [inline src/Codeception/Module/Db.php]: Ahh, wrong example, It should be:

```php
$I->grabColumnFromDatabase('users', 'email', array('name' => 'RebOOter'));
```

**Comment by RebOOter** [comment]
Sorry, my bad
Fixed

**Comment by DavertMik** [comment]
thanks!

**Pull request merged**